File: sdk/typescript/src/phone/resources/transcripts.ts
/**
 * inkbox-phone/resources/transcripts.ts
 *
 * Transcript retrieval.
 */

import { HttpTransport } from "../../_http.js";
import {
  PhoneTranscript,
  RawPhoneTranscript,
  parsePhoneTranscript,
} from "../types.js";

export class TranscriptsResource {
  constructor(private readonly http: HttpTransport) {}

  /**
   * List all transcript segments for a call, ordered by sequence number.
   *
   * @param phoneNumberId - UUID of the phone number.
   * @param callId - UUID of the call.
   */
  async list(
    phoneNumberId: string,
    callId: string,
  ): Promise<PhoneTranscript[]> {
    const data = await this.http.get<RawPhoneTranscript[]>(
      `/numbers/${phoneNumberId}/calls/${callId}/transcripts`,
    );
    return data.map(parsePhoneTranscript);
  }
}

File: sdk/rust/src/phone/resources/transcripts.rs
//! Transcript retrieval.

use std::sync::Arc;

use crate::error::Result;
use crate::http::HttpTransport;
use crate::phone::types::PhoneTranscript;

pub struct TranscriptsResource {
    http: Arc<HttpTransport>,
}

impl TranscriptsResource {
    pub fn new(http: Arc<HttpTransport>) -> Self {
        Self { http }
    }

    /// List all transcript segments for a call, ordered by sequence number.
    ///
    /// # Arguments
    /// * `phone_number_id` - UUID (or string) of the phone number.
    /// * `call_id` - UUID (or string) of the call.
    pub fn list(&self, phone_number_id: &str, call_id: &str) -> Result<Vec<PhoneTranscript>> {
        let data = self.http.get(
            &format!("/numbers/{phone_number_id}/calls/{call_id}/transcripts"),
            crate::http::NO_QUERY,
        )?;
        Ok(serde_json::from_value(data)?)
    }
}
